src/components/App.js

```diff
@@ -1,17 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../styles/App.scss';
 import '../styles/Button.scss';
 import '../styles/ButtonPanel.scss';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
+import calculate from '../logic/calculate';
 
 function App() {
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+  const [result, setResult] = useState('0');
+  const [error, setError] = useState(null);
+
+  const handleClick = (buttonName) => {
+    const data = calculate({
+      total, next, operation, error,
+    }, buttonName);
+
+    setNext(data.next);
+    setOperation(data.operation);
+    setTotal(data.total);
+    setResult(data.next || data.total);
```

Comment:
We can do away with the `if` statement below by:

```
setResult(data.next || data.total || data.error);
```
Since setting state is asynchronous, it's not a problem even if `setError` is happening in the next line.